### PR TITLE
Call nxsem_destroy or nxmutex_destry in the error path

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

1.  Don't check the return value of nxsem_init or nxmutex_init
2. Fix some style issue

## Impact
error handling

## Testing
Pass CI

### CODE REVIEW

File: drivers/wireless/gs2200m.c [hunk 1]
```diff
@@ -889,10 +889,10 @@ static uint16_t _read_data_len(FAR struct gs2200m_dev_s *dev)
...
 
 enum spi_status_e gs2200m_hal_write(FAR struct gs2200m_dev_s *dev,
-                                    const void *data,
+                                    FAR const void *data,
                                     uint16_t txlen)
 {
-  uint8_t *tx = (uint8_t *)data;
+  FAR uint8_t *tx = (FAR uint8_t *)data;
   uint8_t  hdr[8];
   uint8_t  res[8];
```
Comment (pkarashchenko) [inline drivers/wireless/gs2200m.c:897]:
```suggestion
  uint8_t hdr[8];
  uint8_t res[8];
```

File: drivers/wireless/nrf24l01.c [hunk 2]
```diff
@@ -1496,7 +1497,7 @@ int nrf24l01_register(FAR struct spi_dev_s *spi,
   dev->rx_fifo = rx_fifo;
 
   nxmutex_init(&dev->lock_fifo);
-  nxsem_init(&dev->sem_rx, 0, 0);
+  nxsem_init(&(dev->sem_rx), 0, 0);
```
Comment (pkarashchenko) [inline drivers/wireless/nrf24l01.c:1500]: why is this needed?

File: fs/smartfs/smartfs_smart.c [hunk 3]
```diff
@@ -1298,11 +1298,11 @@ static int smartfs_readdir(FAR struct inode *mountpt,
...
-  int                   ret;
-  uint16_t              entrysize;
-  struct                smartfs_chain_header_s *header;
-  struct                smart_read_write_s readwrite;
-  struct                smartfs_entry_header_s *entry;
+  FAR struct smartfs_chain_header_s *header;
+  struct smart_read_write_s readwrite;
+  FAR struct smartfs_entry_header_s *entry;
+  uint16_t entrysize;
+  int      ret;
```
Comment (pkarashchenko) [inline fs/smartfs/smartfs_smart.c:1305]:
```suggestion
  int ret;
```

**Review by pkarashchenko — APPROVED** [review]

File: fs/smartfs/smartfs_smart.c (hunk 3 above)
Comment (xiaoxiang781216) [reply fs/smartfs/smartfs_smart.c]: Done.

File: drivers/wireless/nrf24l01.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/wireless/nrf24l01.c]: Done.

File: drivers/wireless/gs2200m.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/wireless/gs2200m.c:897]: Done.

**Review by pkarashchenko — APPROVED** [review]
